qiskit_experiments/library/randomized_benchmarking/clifford_utils.py

```diff
@@ -121,26 +125,141 @@ def _truncate_inactive_qubits(
     return res
 
 
+def _synthesize_clifford(
+    clifford: Clifford,
+    basis_gates: Optional[Tuple[str]],
+    coupling_tuple: Optional[Tuple[Tuple[int, int]]] = None,
+    synthesis_method: str = DEFAULT_SYNTHESIS_METHOD,
+) -> QuantumCircuit:
+    """Synthesize a circuit of a Clifford element. The resulting circuit contains only
+    ``basis_gates`` and it complies with ``coupling_tuple``.
+
+    Args:
+        clifford: Clifford element to be converted
+        basis_gates: basis gates to use in the conversion
+        coupling_tuple: coupling map to use in the conversion in the form of tuple of edges
+        synthesis_method: conversion algorithm name
+
+    Returns:
+        Synthesized circuit
+    """
+    qc = QuantumCircuit(clifford.num_qubits, name=str(clifford))
+    qc.append(clifford, qc.qubits)
+    return _synthesize_clifford_circuit(
+        qc,
+        basis_gates=basis_gates,
+        coupling_tuple=coupling_tuple,
+        synthesis_method=synthesis_method,
+    )
```

Comment:
I think there's a slight misunderstanding. The extra translations you mentioned happen only when synthesizing a 1- or 2-qubit Clifford using a custom synthesis method. I think it's rare and users who has a Clifford circuit can directly call `_synthesize_clifford_circuit` as I do so [here](https://github.com/Qiskit-Extensions/qiskit-experiments/blob/c116a607dae819836ab831b6f10b9416820d1067/qiskit_experiments/library/randomized_benchmarking/clifford_utils.py#L402). Also `_synthesize_clifford` is used only in the constructor of InterleavedRB, that suggests there is no clear need for its performance improvement.

Somewhat related to this, if `HighLevelSynthesis` had a method (namely `synthesize`?) such that it takes a Clifford and returns a QuantumCircuit, it would be very helpful. Actually, if I had it, I didn't need to wrap a Clifford with a QuantumCircuit just because `HighLevelSynthesis.run` requires a DAGCircuit and support only the QuantumCircuit-to-QuantumCircuit conversion through `PassManager.run`.

Reply:
Somewhat related to this, if `HighLevelSynthesis` had a method such that it takes a Clifford and returns a QuantumCircuit (namely `synthesize`?), it would be very helpful. Actually, if it was, I didn't need to wrap a Clifford with a QuantumCircuit here just because `HighLevelSynthesis.run` requires a DAGCircuit and support only the QuantumCircuit-to-QuantumCircuit conversion through `PassManager.run`.